modules/bibsword/lib/bibsword_client_dblayer.py

```diff
@@ -15,349 +17,390 @@
-        @return: (datetime) datetime of the last update (yyyy-mm-dd hh:mm:ss)
-    '''
-
-    qstr = '''SELECT last_update ''' \
-           '''FROM swrREMOTESERVER ''' \
-           '''WHERE id=%s '''
-    qres = run_sql(qstr, (id_remoteserver, ))
-
-    if len(qres) == 0:
-        return '0'
-
-    return qres[0][0]
-
-
-def get_all_remote_server(id_server):
-    '''
-        This function select the name of all remote service implementing the
-        SWORD protocol. It returns a list of dictionnary containing three fields:
-        id, name and host
-        @return: (remote_server) list of dictionnary (id - name - host) of each
-                  remote server
-    '''
-
-    remote_servers = []
-
-    if id_server == '':
-        qstr = '''SELECT id, name, host FROM swrREMOTESERVER'''
-        qres = run_sql(qstr)
-    else :
-        qstr = ''' SELECT id, name, host FROM swrREMOTESERVER WHERE id=%s'''
-        qres = run_sql(qstr, (id_server, ))
+    query = """
+    INSERT INTO swrCLIENTTEMPSUBMISSION
+                (id, object, last_updated)
+    VALUES      (%s, %s, %s)
+    """
 
+    now = convert_datestruct_to_datetext(time.localtime())
+    params = (sid, sobject_blob, now)
 
-    for res in qres:
-        remote_server = {}
-        remote_server['id'] = res[0]
-        remote_server['name'] = res[1]
-        remote_server['host'] = res[2]
-        remote_servers.append(remote_server)
+    try:
+        res = run_sql(query, params)
+    except Exception:
+        return False
+    else:
+        return True
 
-    return remote_servers
 
+def retrieve_temp_submission(
+    sid
+):
+    """
+    Retrieve the temporary submission.
+    """
 
-def is_record_sent_to_server(id_server, id_record):
-    '''
-        check in the table swrCLIENTDATA that the current record has not already
-        been sent before
-        @param id_server: id of the remote server where to send the record
-        @param id_record: id of the record to send
-        return : True if a value was found, false else
-    '''
+    query = """
+    SELECT  object
+    FROM    swrCLIENTTEMPSUBMISSION
+    WHERE   id=%s
+    """
 
-    qstr = '''SELECT COUNT(*) FROM swrCLIENTDATA ''' \
-           '''WHERE id_swrREMOTESERVER=%s AND id_record=%s ''' \
-           '''AND status NOT LIKE 'removed' '''
-    qres = run_sql(qstr, (id_server, id_record, ))
-
-    if (qres[0][0] == 0):
+    params = (sid,)
+
+    res = run_sql(query, params)
+
+    if res:
+        return res[0][0]
+    else:
+        return None
+
+
+def update_temp_submission(
+    sid,
+    sobject_blob
+):
+    """
+    Update the temporary submission.
+    """
+
+    query = """
+    UPDATE  swrCLIENTTEMPSUBMISSION
+    SET     object=%s,
+            last_updated=%s
+    WHERE   id=%s
+    """
+
+    now = convert_datestruct_to_datetext(time.localtime())
+    params = (sobject_blob, now, sid)
+
+    res = run_sql(query, params)
+
+    return res
+
+
+def delete_temp_submission(
+    sid
+):
```

Comment:
I agree, but it would take too much time to clean it and it would bring absolutely no gain, so I decided to leave it as it is.